### PR TITLE
Added a multi-device example to rocBLAS-Example repository

### PR DESCRIPTION
**Pull request by NaveenElumalaiAMD**
This rocBLAS example showcases the use of multi-devices and multiple streams per device to call the 'rocblas_sgemm' function.

### CODE REVIEW

File: Patterns/Multi-device/Makefile [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2019-2020 Advanced Micro Devices, Inc. All rights reserved.
```
Comment (leekillough) [inline Patterns/Multi-device/Makefile:1]: Test comment

File: Patterns/Multi-device/Multi-device.cpp [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+
+    //Scalar 'alpha' value used for multiplication
+    float h_Alpha = options.alpha;
+
+    //Scalar 'beta' value used for multiplication
+    float h_Beta = options.beta;
+
+    //Edge condition check
+    if(N <= 0)
+    {
```
[the remark this answers is not in the document]
Comment (amcamd) [reply Patterns/Multi-device/Multi-device.cpp:68]: change check to N<0 || M < 0 || K < 0

File: Patterns/Multi-device/Multi-device.cpp [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+            //'start_offset_B' points to the starting address of matrix 'B' in accordance with the NUM_STREAMS and NUM_DEVICES
+            int start_offset_B
+                = device_Id > 0 ? device_Id * NUM_STREAMS * K * N : stream_Id * K * N;
+            start_offset_B
+                = stream_Id > 0 && device_Id > 0 ? start_offset_B * stream_Id : start_offset_B;
+
+            helpers::matIdentity(h_B.data() + start_offset_B, K, N, ldb);
+        }
+    }
+
```
[the remark this answers is not in the document]
Comment (amcamd) [reply Patterns/Multi-device/Multi-device.cpp:145]: I think they can be initialized with
fillVectorNormRand(h_A);
fillVectorNormRand(h_B);
fillVectorNormRand(h_C);
h_Gold = h_C;

**Comment by NaveenElumalaiAMD** [comment]
Thank you for the comments Andrew. I will make the necessary changes.

File: Patterns/Multi-device/Multi-device.cpp [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+    float h_Alpha = options.alpha;
+
+    //Scalar 'beta' value used for multiplication
+    float h_Beta = options.beta;
+
+    //Edge condition check
+    if(K < 0 || M < 0 || N < 0)
+    {
+        std::cout << "The value of 'n' should be greater than zero" << std::endl;
+        return 0;
```
Comment (amcamd) [inline Patterns/Multi-device/Multi-device.cpp:70]: The value of n   ->  The value of m, n, k

**Review by amcamd — APPROVED** [review]
LGTM

**Pull request merged**